src/um/winnt.rs

```diff
@@ -1592,14 +3446,377 @@ ENUM!{enum JOBOBJECTINFOCLASS {
+}}
+STRUCT!{struct SYSTEM_LOGICAL_PROCESSOR_INFORMATION {
+    ProcessorMask: ULONG_PTR,
+    Relationship: LOGICAL_PROCESSOR_RELATIONSHIP,
+    u: SYSTEM_LOGICAL_PROCESSOR_INFORMATION_u,
+}}
+pub type PSYSTEM_LOGICAL_PROCESSOR_INFORMATION = *mut SYSTEM_LOGICAL_PROCESSOR_INFORMATION;
+STRUCT!{struct PROCESSOR_RELATIONSHIP {
+    Flags: BYTE,
+    EfficiencyClass: BYTE,
+    Reserved: [BYTE; 20],
+    GroupCount: WORD,
+    GroupMask: [GROUP_AFFINITY; ANYSIZE_ARRAY],
+}}
+pub type PPROCESSOR_RELATIONSHIP = *mut PROCESSOR_RELATIONSHIP;
+STRUCT!{struct NUMA_NODE_RELATIONSHIP {
+    NodeNumber: DWORD,
+    Reserved: [BYTE; 20],
+    GroupMask: GROUP_AFFINITY,
+}}
+pub type PNUMA_NODE_RELATIONSHIP = *mut NUMA_NODE_RELATIONSHIP;
+STRUCT!{struct CACHE_RELATIONSHIP {
+    Level: BYTE,
+    Associativity: BYTE,
+    LineSize: WORD,
+    CacheSize: DWORD,
+    Type: PROCESSOR_CACHE_TYPE,
+    Reserved: [BYTE; 20],
+    GroupMask: GROUP_AFFINITY,
+}}
+pub type PCACHE_RELATIONSHIP = *mut CACHE_RELATIONSHIP;
+STRUCT!{struct PROCESSOR_GROUP_INFO {
+    MaximumProcessorCount: BYTE,
+    ActiveProcessorCount: BYTE,
+    Reserved: [BYTE; 38],
+    ActiveProcessorMask: KAFFINITY,
+}}
+pub type PPROCESSOR_GROUP_INFO = *mut PROCESSOR_GROUP_INFO;
+STRUCT!{struct GROUP_RELATIONSHIP {
+    MaximumGroupCount: WORD,
+    ActiveGroupCount: WORD,
+    Reserved: [BYTE; 20],
+    GroupInfo: [PROCESSOR_GROUP_INFO; ANYSIZE_ARRAY],
+}}
+pub type PGROUP_RELATIONSHIP = *mut GROUP_RELATIONSHIP;
+UNION2!{union SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX_u {
+    [u8; 68] [u8; 72],
+    Processor Processor_mut: PROCESSOR_RELATIONSHIP,
+    NumaNode NumaNode_mut: NUMA_NODE_RELATIONSHIP,
+    Cache Cache_mut: CACHE_RELATIONSHIP,
+    Group Group_mut: GROUP_RELATIONSHIP,
+}}
+STRUCT!{struct SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX {
+    Relationship: LOGICAL_PROCESSOR_RELATIONSHIP,
+    Size: DWORD,
+    u: SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX_u,
+}}
+pub type PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX = *mut SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX;
+ENUM!{enum CPU_SET_INFORMATION_TYPE {
+    CpuSetInformation,
+}}
+pub type PCPU_SET_INFORMATION_TYPE = *mut CPU_SET_INFORMATION_TYPE;
+pub const SYSTEM_CPU_SET_INFORMATION_PARKED: BYTE = 0x1;
+pub const SYSTEM_CPU_SET_INFORMATION_ALLOCATED: BYTE = 0x2;
+pub const SYSTEM_CPU_SET_INFORMATION_ALLOCATED_TO_TARGET_PROCESS: BYTE = 0x4;
+pub const SYSTEM_CPU_SET_INFORMATION_REALTIME: BYTE = 0x8;
+STRUCT!{struct SYSTEM_CPU_SET_INFORMATION_CpuSet_u_s {
+    BitFields: BYTE,
+}}
+BITFIELD!(SYSTEM_CPU_SET_INFORMATION_CpuSet_u_s BitFields: BYTE [
+    Parked set_Parked[0..1],
+    Allocated set_Allocated[1..2],
+    AllocatedToTargetProcess set_AllocatedToTargetProcess[2..3],
+    RealTime set_RealTime[3..4],
+    ReservedFlags set_ReservedFlags[4..8],
+]);
+UNION2!{union SYSTEM_CPU_SET_INFORMATION_CpuSet_u {
+    [u8; 1],
+    AllFlags AllFlags_mut: BYTE,
+    s s_mut: SYSTEM_CPU_SET_INFORMATION_CpuSet_u_s,
+}}
```

Comment:
Another case where the bitfield should be applied on a field directly without an extra union.